lib/Gocdb_Services/Config.php

```diff
@@ -87,21 +114,113 @@ public function GetSchemaXML() {
     }
 
     /**
-     * Opens the local_info.xml file for reading, returning a simplexml object.
+     * Returns a simplexml object representing the local_info.xml file
      * @return simple xml object
      */
     private function GetLocalInfoXML() {
-        return simplexml_load_file($this->getLocalInfoFileLocation());
+        if ($this->local_info_xml == NULL) {
+            if (!($this->local_info_xml = $this->readLocalInfoXML($this->getLocalInfoFileLocation(), $this->local_info_override))) {
+                throw new \ErrorException("Failed to load xml configuration file: ".$this->getLocalInfoFileLocation());
+            } 
+        } 
+        return $this->local_info_xml;
```

Comment:
So I suppose in some sense, `GetLocalInfoXML` returns what it always returned - a "complete" set of local info. 

It just returns a different set based on the value of `$this->local_info_override`.

Is that right?

Reply:
Correct. In fact, in the xml file there must be one (and only one) <config> element without a url= attribute and this must be 'complete', meaning all sub-elements in the schema must be given a value there.  Defining a <config url= ... > effectively overwrites the values from this 'base' element if the url is matched.